sdk-development-guidelines.md

```diff
@@ -0,0 +1,200 @@
+    * MUST prefix all exported resources (layout files, xml files, drawable files, color ids, dimension ids, string ids, etc.)  with <shortname> + "_" (e.g. "sug_" for a Suggestion Module) using [resourcePrefix](https://google.github.io/android-gradle-dsl/current/com.android.build.gradle.LibraryExtension.html#com.android.build.gradle.LibraryExtension:resourcePrefix) in the gradle build
+* For iOS <img src="images/ios.png" alt="drawing" width="15"/>:
+    * SHOULD follow the standard project layout for [iOS](https://developer.apple.com/documentation/xcode/managing-files-and-folders-in-your-xcode-project) projects.
+    * RECOMMEND to not have source file header comment blocks (such as the Xcode default header comment) as these easily get out of date and the information such as author/version of commits is tracked in SCM
+
+
+### Build
+* Development version MUST build with single command after clean clone, i.e. no local setup necessary
+* MUST NOT depend on locally stored libraries
+* MUST NOT use wildcards (+) for exported dependencies, i.e. dependencies that SDK clients will inherit
+* MUST have a sample app - a minimal app that integrates the SDK and shows how to use the SDK features. It SHOULD be written in Kotlin (Android) or Swift (iOS).
+* For Android <img src="images/android.png" alt="drawing" width="15"/>:
+    * MUST use [gradle](https://docs.gradle.org/current/userguide/userguide.html) with [gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) to build Android library or Java library
+    * MUST have consistent minSdk and supportLib versions with other SDKs
+        * SHOULD use [shared build configuration](https://github.com/rakutentech/android-buildconfig) for consistent version with other SDKs/libraries
+        * MAY use manual configuration with values (same as in [shared build configuration](https://github.com/rakutentech/android-buildconfig)), if so
+            * SHOULD use minSdk 23
+            * SHOULD use latest versions of [Android X](https://developer.android.com/jetpack/androidx) libraries (if required)
+        * Example 1: using the [auto value](https://github.com/google/auto/tree/master/value) annotation processor in version 1.4-rc1 is allowed because it is not exported to SDK clients
+        * Example 2: using [okhttp](https://github.com/square/okhttp) compile/implementatino dependency in version 3.4.0-RC1 is not allowed because that dependency will be imposed onto SDK clients
+* For iOS <img src="images/ios.png" alt="drawing" width="15"/>:
+    * SHOULD be built from a [CocoaPods](https://guides.cocoapods.org/) Podspec or [SPM](https://www.swift.org/package-manager/)
+    * SHOULD use [fastlane](https://docs.fastlane.tools/) to automate tasks and MAY use our [shared build configuration](https://github.com/rakutentech/ios-buildconfig) to help with setup and maintain consistency
+    * SHOULD target iOS 12.0 and above, MAY target a more recent iOS version
```

Comment:
Maybe
> SHOULD set minimum deployment target to a recent iOS version unless there is an engineering or product decision to support older versions

cos mentioning specific versions means this soon goes out of date

Reply:
changed to suggested statement